### PR TITLE
Telemetry configurable Added

### PR DESCRIPTION
**Pull request by ajinkyapandetekdi**

### CODE REVIEW

File: src/environments/environment.ts [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
-    host: "https://telemetry-dev.theall.ai",
-    env: "languagelab.portal", // Required. Unique environment where the event has occured.
+    host: 'https://telemetry-dev.theall.ai',
+    env: 'languagelab.portal', // Required. Unique environment where the event has occured.
     apislug: '/v1/mysql/',
-    endpoint: 'telemetry'
-  }
+    endpoint: 'telemetry',
+  },
+  telemetry_mode: 'ET', // can assign any variables as follows: Essential-Telemetry = ET, Normal-Telemetry = NT, and Detailed-Telemetry = DT.
```
Comment (gouravmore) [inline src/environments/environment.ts:21]: Make the default as DT for now.

File: src/app/login/login.component.ts [hunk 2]
```diff
@@ -49,17 +49,22 @@ export class LoginComponent {
...
-            err: data?.message,
-            errtype: data?.status,
-          });
+          this.telemetryService.error(
+            data?.message,
+            {
+              err: data?.message,
+              errtype: data?.status,
+            },
+            'ET'
```
Comment (gouravmore) [inline src/app/login/login.component.ts:66]: all the errors should Detailed-Telemetry = DT

File: src/app/telemetry.service.ts [hunk 3]
```diff
@@ -41,52 +46,57 @@ export class TelemetryService {
...
+      edata: { type: 'content', mode: 'play', pageid: pageid },
+    });
   }
 
-  public log(type, message, pageid, data) {
-    this.startDuration = new Date().getTime();
-    CsTelemetryModule.instance.telemetryService.raiseLogTelemetry(
-      {
+  public log(type, message, pageid, data, currentMode) {
+    if (
```
Comment (gouravmore) [inline src/app/telemetry.service.ts:81]: Write one common function which will return true/false

File: src/app/telemetry.service.ts [hunk 4]
```diff
@@ -96,71 +106,124 @@ export class TelemetryService {
...
+      options: this.getEventOptions(),
     });
-
   }
 
-  public response(data) {
-    CsTelemetryModule.instance.telemetryService.raiseResponseTelemetry(data, this.getEventOptions());
+  public response(data, currentMode) {
+    if (
+      (this.TELEMETRY_MODE === 'ET' && currentMode === 'ET') ||
```
Comment (gouravmore) [inline src/app/telemetry.service.ts:117]: Write one common function which will return true/false

File: src/app/views/pages/level/level.component.html [hunk 5]
```diff
@@ -17,7 +17,7 @@
             </label>
           </div>
           <div class="col-md-6">
-            <button class="px-3 py-2 rounded bg-primary border-0 text-white fs-10 mt-1" (click)="telemetryService.interact('logout', 'level'); logout()">Logout</button>
+            <button class="px-3 py-2 rounded bg-primary border-0 text-white fs-10 mt-1" (click)="telemetryService.interact('logout', 'level','ET'); logout()">Logout</button>
```
Comment (gouravmore) [inline src/app/views/pages/level/level.component.html:20]: this should be NT

File: src/assets/games/index.html [hunk 6]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+<!doctype html><html lang="en"><head><meta charset="utf-8"/><link rel="icon" href="./logo.png"/><meta name="viewport" content="width=device-width,initial-scale=1"/><meta name="theme-color" content="#000000"/><meta name="description" content="Web site created using create-react-app"/><link rel="apple-touch-icon" href="./logo192.png"/><link rel="manifest" href="./manifest.json"/><title>Netskill-English Learning App</title><script defer="defer" src="./static/js/main.60432f80.js"></script><link href="./static/css/main.58d558c4.css" rel="stylesheet"></head><body><noscript>You need to enable JavaScript to run this app.</noscript><div id="root"></div></body></html>
```
Comment (gouravmore) [inline src/assets/games/index.html:1]: Are there any changes in the games?

File: src/assets/sb-all/static/js/main.7c446933.js.LICENSE.txt [hunk 7]
```diff
@@ -0,0 +1,100 @@
+/*
```
Comment (gouravmore) [inline src/assets/sb-all/static/js/main.7c446933.js.LICENSE.txt:1]: Is this changes required for this PR?

File: src/environments/environment.prod.ts [hunk 8]
```diff
@@ -16,5 +16,6 @@ export const environment = {
     env: "languagelab.portal", // Required. Unique environment where the event has occured.
     apislug: '/v1/mysql/',
     endpoint: 'telemetry'
-  }
+  },
+  telemetry_mode: 'ET' // can assign any variables as follows: Essential-Telemetry = ET, Normal-Telemetry = NT, and Detailed-Telemetry = DT.
```
Comment (gouravmore) [inline src/environments/environment.prod.ts:20]: Default should be DT

**Review by gouravmore — CHANGES_REQUESTED** [review]

File: src/assets/games/index.html (hunk 6 above)
Comment (ajinkyapandetekdi) [reply src/assets/games/index.html:1]: I didn't change anything in those files, but I got those files after taking a pull.

File: src/assets/sb-all/static/js/main.7c446933.js.LICENSE.txt (hunk 7 above)
Comment (ajinkyapandetekdi) [reply src/assets/sb-all/static/js/main.7c446933.js.LICENSE.txt:1]: i took pull then it came to my code. This is only txt file

**Pull request closed without merging**